app/bundles/LeadBundle/EventListener/FormSubscriber.php

```diff
@@ -39,45 +43,71 @@ public function onFormBuilder(FormBuilderEvent $event)
-            'callback'    => '\Mautic\LeadBundle\Helper\FormEventHelper::changeLists',
+            'formType' => 'leadlist_action',
+            'callback' => '\Mautic\LeadBundle\Helper\FormEventHelper::changeLists',
         ];
         $event->addSubmitAction('lead.changelist', $action);
 
         // modify tags
         $action = [
-            'group'             => 'mautic.lead.lead.submitaction',
-            'label'             => 'mautic.lead.lead.events.changetags',
-            'description'       => 'mautic.lead.lead.events.changetags_descr',
-            'formType'          => 'modify_lead_tags',
-            'callback'          => '\Mautic\LeadBundle\Helper\EventHelper::updateTags',
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.changetags',
+            'description' => 'mautic.lead.lead.events.changetags_descr',
+            'formType' => 'modify_lead_tags',
+            'callback' => '\Mautic\LeadBundle\Helper\EventHelper::updateTags',
             'allowCampaignForm' => true,
         ];
         $event->addSubmitAction('lead.changetags', $action);
 
         // add UTM tags
         $action = [
-            'group'       => 'mautic.lead.lead.submitaction',
-            'label'       => 'mautic.lead.lead.events.addutmtags',
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.addutmtags',
             'description' => 'mautic.lead.lead.events.addutmtags_descr',
-            'formType'    => 'lead_action_addutmtags',
-            'formTheme'   => 'MauticLeadBundle:FormTheme\\ActionAddUtmTags',
-            'callback'    => '\Mautic\LeadBundle\Helper\EventHelper::addUtmTags',
+            'formType' => 'lead_action_addutmtags',
+            'formTheme' => 'MauticLeadBundle:FormTheme\\ActionAddUtmTags',
+            'callback' => '\Mautic\LeadBundle\Helper\EventHelper::addUtmTags',
         ];
         $event->addSubmitAction('lead.addutmtags', $action);
+
+        // add Do Not Contact
+        $action = [
+            'group' => 'mautic.lead.lead.submitaction',
+            'label' => 'mautic.lead.lead.events.removedonotcontact',
+            'description' => 'mautic.lead.lead.events.removedonotcontact_descr',
+            'formType' => 'lead_action_removedonotcontact',
+            'formTheme' => 'MauticLeadBundle:FormTheme\\ActionRemoveDoNotContact',
+            'eventName' => FormEvents::FORM_ON_SUBMIT,
+            'allowCampaignForm' => true,
+        ];
+        $event->addSubmitAction('lead.RemoveDoNotContact', $action);
+    }
+
+
+    /**
+     * Trigger event for when a form is submitted.
+     *
+     * @param SubmissionEvent $event
+     */
+    public function onFormSubmit(SubmissionEvent $event)
+    {
+        $form = $event->getResults();
+        if(isset($form['email']) && !empty($form['email'])){
+            //$emailModel->removeDoNotContactEntry($form['email']);
```

Comment:
@alanhartless @escopecz  Could you help me? How Can I call Email Model on this line?